statsmodels/base/data.py

```diff
@@ -504,7 +504,7 @@ class PandasData(ModelData):
     def _convert_endog_exog(self, endog, exog=None):
         #TODO: remove this when we handle dtype systematically
         endog = np.asarray(endog)
-        exog = exog if exog is None else np.asarray(exog)
+        exog = exog if exog is None else np.asarray(exog, dtype=float)
         if endog.dtype == object or exog is not None and exog.dtype == object:
```

Comment:
what happens now if exog.dtype is object?  does numpy raise some casting error?
check in line 508 would not apply anymore

Reply:
Seems to cast it to float, or at least try. 

Should add a test with an uncastable object, and fix the test that is now failing (which is right)